src/runtime/virtcontainers/fs_share_linux.go

```diff
@@ -31,15 +34,26 @@ func unmountNoFollow(path string) error {
 }
 
 type FilesystemShare struct {
-	sandbox *Sandbox
+	sandbox            *Sandbox
+	watcher            *fsnotify.Watcher
+	srcDstMap          map[string]string
+	watcherDoneChannel chan bool
 	sync.Mutex
 	prepared bool
 }
 
 func NewFilesystemShare(s *Sandbox) (FilesystemSharer, error) {
+	watcher, err := fsnotify.NewWatcher()
+	if err != nil {
+		return nil, fmt.Errorf("Failed to add watcher")
```

Comment:
Can we print the original error as well?

Reply:
I'll add the err to the Errorf statement.